src/graph-layouts.jl

```diff
@@ -0,0 +1,89 @@
+export distgraph, distdigraph, num_vertices, vertices, adj
+###
+# Graph data structure definitions.
+###
+
+# Allowed Graph datastructures:
+typealias AdjacencyList Array{Array{Int, 1}, 1}
+typealias AdjacencyMatrix AbstractArray{Int, 2}                      # Covers sparse matrices
+typealias GraphStruct Union{AdjacencyList, AdjacencyMatrix}
+
+
+# Generic (undirected or directed, up to the user to decide) Graph
+type DistGraph{S<:GraphStruct}
```

Comment:
This representation is similar to what Graphs.jl and LightGraphs.jl use. Implementing `convert` methods will allow users to apply algorithms directly on their graphs. Writing algorithms will also be more convenient with a representation such as this. 

Will this cause any performance problems?